invenio_rdm_records/resources/serializers/schemaorg/schema.py

```diff
@@ -205,6 +206,20 @@ class SchemaorgSchema(BaseSerializerSchema, CommonFieldsMixin):
 
     url = fields.Method("get_url")
 
+    # Fields that are specific to certain resource types.
+    measurementTechnique = fields.Method("get_measurement_techniques")
+    distribution = fields.Method("get_distribution")
+
+    @pre_dump
+    def hide_closed_files(self, obj, many, **kwargs):
+        """Hide files if the record is not Open Access."""
+        ret = deepcopy(obj)
+        access = py_.get(ret, "access.files")
+        files = py_.get(ret, "files")
+        if access == "restricted" and files:
+            del ret["files"]
+        return ret
```

Comment:
minor/comment: aren't the files already removed by the service schema dump in case the record has restricted files?

Reply:
They are removed if the user has no permission to view the files, however if the user has permission to see the files they will not be removed by the service shema but still we don't want them to be shown in the jsonld (unless I am missing something)

Reply:
We want to keep it consistent throughout serializers no matter the end-use (like e.g. in the "regular" InvenioRDM JSON response), so it's ok to show the files to a user that has permission to see them.

Reply:
Ok, just to confirm in "old" Zenodo we were also hiding them, see here the [code](https://github.com/zenodo/zenodo/blob/2dbd8ca0ddf9f9ee8b73a7d8e91e6157ce187fab/zenodo/modules/records/serializers/schemas/schemaorg.py#L284-L289). Please confirm that now we don't need this anymore and I will remove it (sorry for being so annoying, just want to make sure we don't need to revert the change :sweat_smile: )

Reply:
Discussed IRL, we will remove it